syntax = "proto3";
import "google/protobuf/timestamp.proto";
import "google/protobuf/wrappers.proto";


package protos.accounts;

enum SSOProvider {
  UNKNOWN = 0;
  OKTA = 1;
}

message AccountApiToken {
  string key = 1;
  string created_by = 2;
  google.protobuf.Timestamp created_at = 3;
  uint64 account_id = 4;
}

message UserFlags {
  google.protobuf.BoolValue is_email_verified = 1;
  google.protobuf.BoolValue is_account_owner = 2;
}

message User {
  uint64 id = 1;
  string email = 2;
  string first_name = 3;
  string last_name = 4;
  UserFlags user_flags = 5;
}
